link/syscalls.go

```diff
@@ -113,9 +115,7 @@ var haveProgQuery = internal.NewFeatureTest("BPF_PROG_QUERY", "4.15", func() err
 	}
 
 	err := sys.ProgQuery(&attr)
-	if errors.Is(err, unix.EINVAL) {
-		return internal.ErrNotSupported
```

Comment:
This test never returns ErrNotSupported now? I guess it's technicall valid to say "I can only detect presence of a feature, not absence" but is that so useful really?

This probe is a bit weird since the "feature present" case is also an error. So the expected matrix of outcomes is:

* EBADF: syscall is supported
* nil: something weird has happened which we don't understand. This should be an error that isn't `ErrNotSupported`.
* other errors (currently EINVAL): syscall is probably not supported. This could be `ErrNotSupported`.

If you want to get rid of EINVAL here I'd do the following:

```
if errors.Is(EBADF) { return nil }
if err != nil { return ErrNotSupported }
return errors.New("syscall succeeded unexpectedly")
```

Reply:
I like this approach because it covers all cases. +1 on returning an error for the `nil` case